txpipe/utils/number_density_stats.py

```diff
@@ -32,14 +38,15 @@ def add_data(self, shear_data, shear_bin):
 
     def collect(self):
         # Get the basic shear numbers - means, counts, variances
-        variances = np.zeros((self.nbin_source, 2))
-        means = np.zeros((self.nbin_source, 2))
+        nb = self.nbin_source
+        variances = np.zeros((nb + 1, 2))
```

Comment:
why is it +1? in the loop below looks like you only fill nb elements?

Reply:
yep, but then we fill in the final one just afterwards with the 2D bit.